src/main/java/net/fabricmc/mapping/tree/TinyMappingFactory.java

```diff
@@ -21,10 +22,32 @@
 	 * @throws IOException           if the reader throws one
 	 * @throws MappingParseException if there is an issue with the v2 format
 	 */
-	public static TinyMapping load(BufferedReader reader) throws IOException, MappingParseException {
+	public static TinyTree load(BufferedReader reader) throws IOException, MappingParseException {
 		Visitor visitor = new Visitor();
 		TinyV2Factory.visit(reader, visitor);
-		return new Mapping(visitor.metadata, visitor.classNames, visitor.classes);
+		return new Tree(visitor.metadata, visitor.classNames, visitor.classes);
+	}
+
+	/**
+	 * Loads a tree model from a buffered reader and automatically determine the input type.
+	 *
+	 * @param reader the buffered reader
+	 * @return the built reader model
+	 * @throws IOException           if the reader throws one
+	 * @throws MappingParseException if there is an issue with the v2 format
+	 */
+	public static TinyTree loadWithDetection(BufferedReader reader) throws IOException, MappingParseException {
```

Comment:
I was thinking of having just one public `load` function that does this, and let everyone forget that there's even 2 different tiny formats.